src/libsync/networkjobs/resources.cpp

```diff
@@ -0,0 +1,118 @@
+#include "resources.h"
+
+#include <QDir>
+#include <QFile>
+#include <QIcon>
+
+namespace OCC {
+
+Q_LOGGING_CATEGORY(lcResources, "sync.networkjob.resource")
+
+namespace {
+    QString hashMd5(const QString &data)
+    {
+        QCryptographicHash hash(QCryptographicHash::Algorithm::Md5);
+        hash.addData(data.toLatin1());
+        return QString::fromLatin1(hash.result().toHex());
+    }
+
+    QString cacheKey(QNetworkReply *reply)
+    {
+        QString urlHash = hashMd5(reply->url().toString());
+        QString eTagHash = hashMd5(reply->header(QNetworkRequest::ETagHeader).toString());
+        return QStringLiteral("%1-%2").arg(urlHash, eTagHash);
```

Comment:
In the context of qml we might require a local file path.
It would be helpful if you would store the files with their file extension.

Reply:
https://doc.qt.io/qt-5/qmimedatabase.html